tools/windows/install-help/cal/TargetMachine.cpp

```diff
@@ -82,19 +82,10 @@ DWORD TargetMachine::DetectMachineType()
     DWORD status = NetServerGetInfo(nullptr, 101, reinterpret_cast<LPBYTE *>(&serverInfo));
     if (status != NERR_Success)
     {
-        /*
-         * If the function fails, the return value can be one of the following error codes.
-         *   - ERROR_ACCESS_DENIED
-         *  The user does not have access to the requested information.
-         *  -  ERROR_INVALID_LEVEL
-         *  The value specified for the level parameter is invalid.
-         *  - ERROR_INVALID_PARAMETER
-         *  The specified parameter is invalid.
-         *  - ERROR_NOT_ENOUGH_MEMORY
-         *  Insufficient memory is available.
-         */
-        WcaLog(LOGMSG_STANDARD, "Failed to get server info: %d %d", status, GetLastError());
-        return status;
+        // NetServerGetInfo failed. This is most likely due to the Server service not running, which we can safely ignore.
```

Comment:
A couple of things
1) we're assuming workstation; however, the customer ticket that generated this is running on server 2019, so  that's not a good assumption
2) we're making this assumption on all errors.  At minimum, we should only do this on the specific error code indicating that the server service isn't running, not all errors.

Reply:
👍 Let's make sure we understand the impact of mis-identifying the machine type. If it has a significant impact, it sounds indeed necessary to scope on specific error types (and even in these cases, we should understand how a mis-identification affects the install logic downstream)

Reply:
I don't think you can make this assumption

> I think it's safe to assume that any server in a Domain environment will have the Server service running as the Netlogon service, which is used to authenticate users in the domain, depends on it.

You have no way of knowing (at least with this code) if this is a transient situation where something else is broken (which is then going to be compounded by our install making the wrong decision) or a permanent state (in which case you can likely treat it the same as a workstation.

And the original comment is that there's a specific error code when the `server` service isn't running, whereas this PR treats _all_ errors as that case.